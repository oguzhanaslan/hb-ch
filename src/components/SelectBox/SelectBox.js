import React from 'react';
import PropTypes from 'prop-types';
import './SelectBox.scss';

const SelectBox = ({ value, onChange }) => {
  return (
    <div>
      <select className='form-select' value={value} onChange={(e) => onChange(e)}>
        <option value='lastAdded'>Last Added</option>
        <option value='mostVoted'>Most Voted (a->z)</option>
        <option value='lessVoted'>Less Voted (z->a)</option>
      </select>
    </div>
  );
};

SelectBox.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default SelectBox;
