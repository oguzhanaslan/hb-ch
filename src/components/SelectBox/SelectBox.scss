.form-select {
  background: #ffffff;
  overflow: hidden;
  border: 1px solid #cbcbcb;
  border-radius: 3px;
  box-shadow: 0 0 1px rgba(0, 0, 0, 0.1), inset 0 0 10px rgba(0, 0, 0, 0.07);
  height: 50px;
  margin: 20px 0;
  font-size: 16px;
  transition: all .2s ease-in;

  &:focus,
  &:hover {
    box-shadow: 0 0 15px rgba(0, 0, 0, 0.2);
  }

  &:after {
    content: 'â–¾';
    position: absolute;
    top: 0;
    right: 12px;
    z-index: 1;
    color: #bbbbbb;
    line-height: 40px;
    font-size: 14px;

    .ie & {
      display: none;
    }
  }
}

select {
  position: relative;
  background: transparent;
  background-image: none;
  display: block;
  width: 100%;
  outline: none;
  color: #888888;
  cursor: pointer;
  z-index: 2;
  border: none;
  padding: 10px 10px 10px 5px;
  // This is so hacky.. (for Firefox)
  // Removes the default select arrow.
  text-overflow: '';
  text-indent: 5px;
  // End of brutal hack

  &[multiple=multiple] {
    height: auto;
    border: 1px solid #cbcbcb;
    border-radius: 3px;
    box-shadow: 0 0 1px rgba(0, 0, 0, 0.1), inset 0 0 10px rgba(0, 0, 0, 0.07);

    &:hover {
      border-color: #cccccc;
      box-shadow: 0 0 15px rgba(0, 0, 0, 0.2);
    }
  }

  option[disabled] {
    color: #eeeeee;
    text-shadow: none;
    border: none;
  }
}

select:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #888888;
}

select::-ms-expand {
  display: none;
}

/*==========  Generic styling  ==========*/

.column-6 {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -20px 0 0 -150px;
  width: 300px;
}
