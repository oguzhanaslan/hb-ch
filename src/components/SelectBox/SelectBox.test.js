import React from 'react';
import { shallow } from 'enzyme';
import SelectBox from './SelectBox';

describe('SelectBox', () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<SelectBox />)));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <select />', () => {
    expect(wrapper.find('select').length).toEqual(1);
  });

  it("should render 3 <option />'s", () => {
    expect(wrapper.find('option').length).toEqual(3);
  });
});
