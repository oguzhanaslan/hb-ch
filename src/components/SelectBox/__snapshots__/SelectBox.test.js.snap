// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        className="form-select"
        onChange={[Function]}
      >
        <option
          value="lastAdded"
        >
          Last Added
        </option>
        <option
          value="mostVoted"
        >
          Most Voted (a-&gt;z)
        </option>
        <option
          value="lessVoted"
        >
          Less Voted (z-&gt;a)
        </option>
      </select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="lastAdded"
          >
            Last Added
          </option>,
          <option
            value="mostVoted"
          >
            Most Voted (a-&gt;z)
          </option>,
          <option
            value="lessVoted"
          >
            Less Voted (z-&gt;a)
          </option>,
        ],
        "className": "form-select",
        "onChange": [Function],
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Last Added",
            "value": "lastAdded",
          },
          "ref": null,
          "rendered": "Last Added",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Most Voted (a->z)",
            "value": "mostVoted",
          },
          "ref": null,
          "rendered": "Most Voted (a->z)",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Less Voted (z->a)",
            "value": "lessVoted",
          },
          "ref": null,
          "rendered": "Less Voted (z->a)",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          className="form-select"
          onChange={[Function]}
        >
          <option
            value="lastAdded"
          >
            Last Added
          </option>
          <option
            value="mostVoted"
          >
            Most Voted (a-&gt;z)
          </option>
          <option
            value="lessVoted"
          >
            Less Voted (z-&gt;a)
          </option>
        </select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="lastAdded"
            >
              Last Added
            </option>,
            <option
              value="mostVoted"
            >
              Most Voted (a-&gt;z)
            </option>,
            <option
              value="lessVoted"
            >
              Less Voted (z-&gt;a)
            </option>,
          ],
          "className": "form-select",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Last Added",
              "value": "lastAdded",
            },
            "ref": null,
            "rendered": "Last Added",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Most Voted (a->z)",
              "value": "mostVoted",
            },
            "ref": null,
            "rendered": "Most Voted (a->z)",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Less Voted (z->a)",
              "value": "lessVoted",
            },
            "ref": null,
            "rendered": "Less Voted (z->a)",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
